### PR TITLE
The category attribute of outlines will now be exported

### PR DESCRIPTION
**Pull request by PatrickGotthard**
Hi @mishako,

would you like to review this code and to validate my implementation against the spec?
This implementation should solve #3

Regards,
Patrick

### CODE REVIEW

**Comment by mishako** [comment]
Good code! Category attribute construction looks slightly wrong to me though. It has to be a comma-separated list as in this example from the spec: `/Harvard/Berkman,/Politics`. Categories themselves can contain slashes indicating their hierarchy. In the example from the spec, Berkman is a subcategory of Harvard (The Berkman Center is part of the Harvard University). I don't think we should care about slashes though, because category can not contain any slashes.

File: src/test/java/com/rometools/opml/io/impl/OPML20GeneratorTest.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+import com.rometools.opml.feed.opml.Opml;
+import com.rometools.opml.feed.opml.Outline;
+import com.rometools.rome.io.FeedException;
+import com.rometools.rome.io.WireFeedOutput;
+
+public class OPML20GeneratorTest {
+
+    @Test
+    public void testCategoryOutput() throws IllegalArgumentException, FeedException, SAXException, IOException, XpathException {
```
Comment (mishako) [inline src/test/java/com/rometools/opml/io/impl/OPML20GeneratorTest.java]: How about simply `throws Exception`? Same below.

File: src/test/java/com/rometools/opml/io/impl/OPML20GeneratorTest.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        checkCategoryOutput(null, "");
+        checkCategoryOutput(Arrays.asList("category1"), "/category1");
+        checkCategoryOutput(Arrays.asList("category1", "category2"), "/category1/category2");
```
Comment (mishako) [inline src/test/java/com/rometools/opml/io/impl/OPML20GeneratorTest.java]: Would be nice to have each test case in a separate method.

File: src/test/java/com/rometools/opml/io/impl/OPML20GeneratorTest.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    }
+
+    private void checkCategoryOutput(final List<String> categories, final String asserted)
```
Comment (mishako) [inline src/test/java/com/rometools/opml/io/impl/OPML20GeneratorTest.java]: Maybe `expected` instead of `asserted`?

**Comment by PatrickGotthard** [comment]
Hi mishako,

thanks for your review, I have implemented all your proposals. I slighty misunderstood the specification regarding the delimiter. The corrected version makes much more sense :)

Regards,
Patrick

**Pull request merged**